frontend/src/app/pages/exercises/exercise/shared/exercise-statistics/shared/decimate-statistics.ts

```diff
@@ -0,0 +1,23 @@
+const maximumNumberOfValues = 200;
+
+export function decimateStatistics<T>(statistic: T[]): T[] {
+    const n = Math.ceil(statistic.length / (maximumNumberOfValues / 2));
+    // Performance shortcut
+    if (n === 1) {
+        return statistic;
+    }
+    // TODO: Try to always use the maximum possible number of values
```

Comment:
Not in this pull request. This is basically an open-ended heuristic optimization.